lib/decorate.js

```diff
@@ -9,6 +9,14 @@ function decorate (name, fn, dependencies) {
     checkDependencies(this, dependencies)
   }
 
+  if (typeof fn.getter === 'function' || typeof fn.setter === 'function') {
+    Object.defineProperty(this, name, {
+      get: fn.getter,
+      set: fn.setter
+    })
+    return this
+  }
+
   this[name] = fn
```

Comment:
This line should probably be inside an `else` block, or am I missing something?

Reply:
Are you missing the `return` statement in the `if` block?

Reply:
Definitely :P